modules/dashboards/library/Dashboards/Form/DashletForm.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Icinga\Module\Dashboards\Form;
+
+use Icinga\Module\Dashboards\Model\Database;
+use ipl\Sql\Select;
+use ipl\Web\Compat\CompatForm;
+
+class DashletForm extends CompatForm
+{
+    use Database;
+
+    public function fetchDashboards()
+    {
+        $dashboards = [];
+
+        $select = (new Select())
+            ->columns('*')
+            ->from('dashboard');
+
+        $data = $this->getDb()->select($select);
+
+        foreach ($data as $dashboard) {
+            $dashboards[$dashboard->id] = $dashboard->name;
+        }
+
+        return $dashboards;
+    }
+
+    public function fetchNewDashboardId()
+    {
+        $dashboardIds = [];
+
+        $data = ['name' => $this->getValue('new_dashboard')];
+
+        $this->getDb()->insert('dashboard', $data);
+
+        $newDashboard = (new Select())
```

Comment:
You don't have to select the generated ID on your own. Since `Sql/Connection` wraps `PDO` you may just call `$db->lastInsertId()`